stdlib/database/sql/src/main/java/org/ballerinalang/database/sql/statement/UpdateStatement.java

```diff
@@ -97,8 +99,16 @@ public Object execute() {
                 generatedKeys = new MapValueImpl<>();
             }
             return createFrozenUpdateResultRecord(count, generatedKeys);
-        } catch (Throwable e) {
-            return SQLDatasourceUtils.getSQLConnectorError(e, "execute update failed: ");
+        } catch (SQLException e) {
+            return SQLDatasourceUtils.getSQLDatabaseError(e, "execute update failed: ");
+            //handleErrorOnTransaction(context);
+           // checkAndObserveSQLError(context, "execute update failed: " + e.getMessage());
+        }  catch (DatabaseException e) {
+            return SQLDatasourceUtils.getSQLDatabaseError(e, "execute update failed: ");
+            //handleErrorOnTransaction(context);
+            // checkAndObserveSQLError(context, "execute update failed: " + e.getMessage());
```

Comment:
Above two catch blocks has the same logic

Reply:
But In both catch I'm calling getSQLDatabaseError method with different exception. So it is calling the overloaded getSQLDatabaseError method